scripts/ebeam_scan_to_multiprobe.py

```diff
@@ -0,0 +1,117 @@
+    n = coordinates.shape[0]
+    if n < 2:
+        raise ValueError("n should be higher than or equal to 2, cannot find deflection angle for n is {}".format(n))
+    elif n == 2:
+        x = coordinates[1, :] - coordinates[0, :]
+        # Determine the angle of this line to the x-axis, -pi/2 < phi < pi/2
+        phi = numpy.arctan2(x[0], x[1])
+    else:
+        A = numpy.hstack((coordinates, numpy.ones((n, 1))))
+
+        # Solve the equation A*x = 0; i.e. find the null space of A. The solution is
+        # the eigenvector corresponding to the smallest singular value.
+        U, s, V = numpy.linalg.svd(A, full_matrices=False)
+        x = numpy.transpose(V[-1])
+
+        # Determine the angle of this line to the x-axis, -pi/2 < phi < pi/2
+        phi = numpy.arctan2(-x[0], x[1])
+    phi = (phi + math.pi / 2) % math.pi - math.pi / 2
+    return phi
+
+
+def get_sem_rotation(ccd):
+    """
+
+    ccd: a camera object
+    return:
+    phi: rotation that needs to be added to the scan rotation in the SEM UI.
+    """
+    image = ccd.data.get(asap=False)
+    n_spots = (8, 8)
+    spot_coordinates, translation, scaling, rotation = FindGridSpots(image, n_spots)
+    coordinates = [translation]
+    input('Press enter after applying x-shift to electron beam: \n')
+    while True:
+        image = ccd.data.get(asap=False)
+        spot_coordinates, translation, scaling, rotation = FindGridSpots(image, n_spots)
+        coordinates.append(translation)
+        inp = input('Applied x-shift to electron beam again? y/n \n')
+        while inp.lower() not in ['y', 'n']:
+            inp = input('{} not an option please choose from y/n \n'.format(inp))
+        if inp.lower() == "n":
+            break
+
+    deflection_angle = get_deflection_angle(numpy.array(coordinates))
+    sem_rotation = deflection_angle - rotation
+    return sem_rotation
+
+
+def main(args):
+    """
+    Handles the command line arguments
+    args is the list of arguments passed
+    return (int): value to return to the OS as program exit code
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--debug', action="store_true", default=False)
+    parser.add_argument("--role", dest="role", metavar="<component>", help="display and update an image on the screen")
+
+    options = parser.parse_args(args[1:])
+    if options.debug:
+        logging.getLogger().setLevel(logging.DEBUG)
+    else:
+        logging.getLogger().setLevel(logging.WARNING)
+
+    if options.role:
+        if get_backend_status() != BACKEND_RUNNING:
+            raise ValueError("Backend is not running while role command is specified.")
+        ccd = model.getComponent(role=options.role)
+    else:
+        ccd = ueye.Camera("camera", "ccd", device=None)
+
+    try:
+        sem_rotation = get_sem_rotation(ccd)
+        print("Apply {:.3f}° to the scan rotation using SEM PC".format(math.degrees(sem_rotation)))
```

Comment:
I'd suggest to add an `except Execption`, which log the whole backtrace, and return 128.